### ISSUE TITLE
Could you please upload the model for horses?

### ISSUE THREAD

**Issue by buaaswf** [jiashunwang/Neural-Pose-Transfer#6]
Dear @TianyunZ @walsvid @jiashunwang 
I just run the demo.py with two horses. The model for a horse is different from the model link you provided.
Could you please upload the model for horses?

Thanks!

**Comment by walsvid** [comment]
Hi @buaaswf ,  if you want to test the horse dataset please use `maxpool.model` as ckpt file, and you also need to change the model in the `demo.py` to the network structure defined in `model_maxpool.py`.

**Comment by buaaswf** [comment]
Dear @walsvid ,
I have tested the maxpool.model with the two horse meshes in this dataset 'Robert W Sumner and Jovan Popovic. Deformation transfer ´ for triangle meshes. ACM Transactions on graphics (TOG), 23(3):399–405, 2004.' 
I changed the points to 8431 to fit for the meshes vertexes. However, the results seem strange, could you please send me two horse test cases? or you call add me on the wechart 18810318179

**Comment by jiashunwang** [comment]
Hi @buaaswf ,
the pretrained model was trained using human meshes and works for different human meshes. If you want to try the horse and camel models, you can use the datasest from 'Robert W Sumner and Jovan Popovic. Deformation transfer ´ to train a new model as we mentioned in the supplementary material.

**Comment by beyondsong** [comment]
Hi @jiashunwang,
Thank you for your good work. Could you please tell me how you expanded the animal dataset since there is little data in it?
Thank you.

**Comment by jiashunwang** [comment]
Hi @beyondsong,
Taking the pair of camel and horse as an example, I took about 10 poses as test data and use the others to train and the result is good. If you want to train a better animal version, I recommend you this dataset[SMAL](https://smal.is.tue.mpg.de/), maybe helpful.

**Comment by beyondsong** [comment]
Hi @jiashunwang,
Thank you for your reply. I have checked with the animal dataset. They provide 48 gallop horse/camel pairs and 53 horse/camel collapse pairs besides 10 pose pairs. Do you mean you train the model using these 101 pairs?
By the way, I want to ask how much the minimum loss in your training process. I get a minimum loss about 0.00013 using your data and the results of pose transfer are not very good. Thank you!

**Comment by jiashunwang** [comment]
Hi @beyondsong 
You are welcom.
For animal dataset, I trained with less than 100 pairs as I remembered. 
And I think the minimum loss is a little bit less than 0.0001. So I guess the problem may happen during testing, may you check another issue about the evaluation metric? We discussed some mistakes may happen. Or would you provide more details about the test results and maybe I can help you with more details.

**Comment by beyondsong** [comment]
Hi @jiashunwang 
I just use your train.py to train the model and demo.py to generate the new .obj. I try to evaluate it using your evaluate.py. And I get 0.0001+ loss using your original.model while getting 0.043 loss using my trained model(Its loss is about 0.00013 when training). And the generated mesh using my model doesn't show a human look.

By the way, there may be something wrong in your evaluate.py. 
![image](https://user-images.githubusercontent.com/32241974/97071130-a21c8d00-1610-11eb-9efc-104868016da0.png)

**Comment by jiashunwang** [comment]
Sorry, I have fix the problem. I think for testing, maybe you can try to shuffle the identity mesh and pose mesh, but make sure the orders of the points of identity and gt are the same.

**Comment by jiashunwang** [comment]
Hi @beyondsong 
Sorry, I have fixed the problem. I think for testing original method, maybe you can try to shuffle the identity mesh and pose mesh, and make sure the orders of the points of identity and gt are the same.

**Comment by beyondsong** [comment]
Hi @jiashunwang 
Yes, I do as you say so it works well when using the original.model that you provided. 
However, when I train the model myself, the training loss is about 0.00013 while the loss is about 0.043, and the generated mesh very strange when I evaluate the model.

**Comment by mikecheninoulu** [comment]
> Hi @jiashunwang
> Yes, I do as you say so it works well when using the original.model that you provided.
> However, when I train the model myself, the training loss is about 0.00013 while the loss is about 0.043, and the generated mesh very strange when I evaluate the model.

Hi, could you provide the training set size and epoch?

**Comment by jiashunwang** [comment]
Hi @beyondsong 
I have asked another user who uses this code to train and it works, so could you provide more details about your settings？

**Comment by beyondsong** [comment]
Hi @jiashunwang 
I just use your original setting to train the code.  Batch size=8, epoch=1000.

**Comment by beyondsong** [comment]
> Hi @beyondsong
> I have asked another user who uses this code to train and it works, so could you provide more details about your settings？

Hi @jiashunwang 
This is the result that generated using my trained model with training loss is 0.00013. However, when I try to test with it, the loss is about 0.04 and the generated mesh is not good.
![1603715647(1)](https://user-images.githubusercontent.com/32241974/97173098-ff5c3e00-17ca-11eb-9e8d-f6df54bc1d9b.png)

**Comment by mikecheninoulu** [comment]
> > Hi @beyondsong
> > I have asked another user who uses this code to train and it works, so could you provide more details about your settings？
> 
> Hi @jiashunwang
> This is the result that generated using my trained model with training loss is 0.00013. However, when I try to test with it, the loss is about 0.04 and the generated mesh is not good.
> ![1603715647(1)](https://user-images.githubusercontent.com/32241974/97173098-ff5c3e00-17ca-11eb-9e8d-f6df54bc1d9b.png)

@beyondsong Hi, I met the same problem, I think the problem is that you are training with original model while testing it with max_pool model

**Comment by beyondsong** [comment]
Hi @mikecheninoulu 
Thank you. I have checked it. I test using my trained model and it is 'from model import NPT' not 'from model_maxpool import NPT' in my test.py. So I think it may not be the problem.

**Comment by jiashunwang** [comment]
Hi,
@beyondsong I think it may still the problem about shuffle and random points, would you try to check the result without faces which means just the points to check if the point cloud is correct visually? And I recommend epoch=200~300, as the paper mentioned.

**Comment by beyondsong** [comment]
Hi @mikecheninoulu @jiashunwang 
Thank you for your advice. I just find that I trained with the maxpool model and test with the original model. It works now.
